File: EjemploH.java
// Parent class
class Animal {
    protected String name;
    public Animal(String name) {
        this.name = name;
    }
    public void eat() {
        System.out.println("The animal is eating");
    }

    public String getName(){
        return name;
    }
}

// Child class inheriting from Animal
class Dog extends Animal {
    public Dog(String nameD) {
        super.name=nameD;
    }
    public void bark() {
        System.out.println("The dog is barking");
    }
}

// Child class inheriting from Animal
class Cat extends Animal {
    public Cat(String nameC) {
        super.name=nameC;
    }
    public void meow() {
        System.out.println("The cat is meowing");
    }
}

// Main class
class Main {
    public static void main(String[] args) {
        Dog myDog = new Dog("Perro");
        myDog.eat(); // Output: The animal is eating
        myDog.bark(); // Output: The dog is barking

        Cat myCat = new Cat("Gato");
        myCat.eat(); // Output: The animal is eating
        myCat.meow(); // Output: The cat is meowing
    }
}

File: EjemploH.py
# Define a parent class
class Animal:
    def __init__(self, name, species):
        self.name = name
        self._species = species

    def make_sound(self):
        pass

# Define a child class that inherits from Animal
class Dog(Animal):
    def __init__(self, name, breed):
        super().__init__(name, species="Dog")
        self.breed = breed

    def make_sound(self):
        return "Woof!"

class Cat(Animal):
    def __init__(self,name,breed):
        super().__init__(name,species="Cat")
        self.breed = breed


if __name__ == "__main__":
    myDog = Dog("Fido", "Labrador")
    myCat=Cat("Garfield","Persa")

    animales=[myDog,myCat]

    for animal in animales:
        print(animal.name)
        print(animal._species)
        print(animal.make_sound())


        '''
        # Access attributes and methods of the parent and child classes
        print(my_dog.name)      # Output: Fido
        print(my_dog.species)   # Output: Dog
        print(my_dog.breed)     # Output: Labrador
        print(my_dog.make_sound())  # Output: Woof!
        '''